File: java/bank-simulation/src/main/java/com/nursyah/bank_simulation/features/auth/IAuthService.java
package com.nursyah.bank_simulation.features.auth;

import com.nursyah.bank_simulation.features.auth.dto.RequestChangePassword;
import com.nursyah.bank_simulation.features.auth.dto.RequestLogin;
import com.nursyah.bank_simulation.features.auth.dto.RequestRegister;
import com.nursyah.bank_simulation.features.auth.dto.RequestResetPassword;
import com.nursyah.bank_simulation.features.auth.dto.RequestToken;
import com.nursyah.bank_simulation.shared.dto.ResponseDto;
import com.nursyah.bank_simulation.shared.dto.ResponseUser;
import java.util.concurrent.CompletableFuture;

public interface IAuthService {
    CompletableFuture<ResponseDto> loginAsync(RequestLogin request);
    CompletableFuture<ResponseDto> registerAsync(RequestRegister request);
    CompletableFuture<ResponseDto> logoutAsync();
    CompletableFuture<ResponseUser> getAuthenticatedUserAsync();
    CompletableFuture<ResponseDto> requestPasswordResetAsync(RequestResetPassword request);
    CompletableFuture<ResponseDto> validatePasswordResetTokenAsync(RequestToken request);
    CompletableFuture<ResponseDto> changePasswordAsync(RequestToken requestToken, RequestChangePassword request);
}

File: dotnet/bank_simulation/Features/Auth/IAuthService.cs
using bank_simulation.Features.Auth.Dto;
using bank_simulation.Shared.Dto;

namespace bank_simulation.Features.Auth;

public interface IAuthService
{
    Task<ResponseDto> LoginAsync(RequestLogin request);
    Task<ResponseDto> RegisterAsync(RequestRegister request);
    Task<ResponseDto> LogoutAsync();
    Task<ResponseUser> GetAuthenticatedUserAsync();
    Task<ResponseDto> RequestPasswordResetAsync(RequestResetPassword request);
    Task<ResponseDto> ValidatePasswordResetTokenAsync(RequestToken request);
    Task<ResponseDto> ChangePassworAsync(RequestToken requestToken, RequestChangePassword request);
}